src/Controls/src/Core/Platform/Android/Extensions/ViewExtensions.cs

```diff
@@ -38,5 +40,34 @@ internal static void MaybeRequestLayout(this AView view)
 			if (!view.IsInLayout && !view.IsLayoutRequested)
 				view.RequestLayout();
 		}
+
+		internal static void AddRecyclerItemViewAccessibility(this AView view, bool shouldBehaveLikeButton)
+		{
+			if (view is null)
+				return;
+
+			var currentDelegate = ViewCompat.GetAccessibilityDelegate(view);
+
+			// If delegate already set correctly, skip
+			if (currentDelegate is ControlsAccessibilityDelegate sid && sid.ShouldBehaveLikeButton == shouldBehaveLikeButton)
+				return;
+
+			// Set new delegate based on existing state
+			if (currentDelegate is ControlsAccessibilityDelegate cad)
+			{
+				cad.ShouldBehaveLikeButton = shouldBehaveLikeButton;
+			}
+			else
+			{
+				var controlsDelegate = new ControlsAccessibilityDelegate(currentDelegate)
+				{
+					ShouldBehaveLikeButton = shouldBehaveLikeButton
+				};
+				ViewCompat.SetAccessibilityDelegate(view, controlsDelegate);
+			}
+
+			// Inform the system that accessibility state has changed
+			view.SendAccessibilityEvent(AViewAccessibility.EventTypes.WindowStateChanged);
```

Comment:
I cannot actually see OnInitializeAccessibilityNodeInfo taking place when I try to debug, so I'm not sure if this way of triggering the AccessibilityEvent is overkill or if there is a better way to do this part.